static/js/fit.js

```diff
@@ -731,6 +775,114 @@ $( document ).ready(function() {
     }
   });
 
+  delegate('textarea[data-course-edit]', 'change', (e, t) => {
+    let key = e.target.id;
+    let formData = {};
+    formData[key] = e.target.value;
+    post(e.target.form.action, formData)
+  });
+
+  delegate('input[data-course-edit]', 'change', (e, t) => {
+    let key = e.target.id;
+    let formData = {};
+    if (e.target.files) {
+      formData['file'] = e.target.files[0];
+    }
+    formData[key] = e.target.value;
+    post(e.target.formAction, formData)
+  });
+
+  delegate('[data-fit-perm-group-type]', 'click', (e, t) => {
+    var group = t.dataset.fitGroupName;
+    let value = '';
+    if (group) {
+    } else {
+      value = (!t.querySelector('input[type=checkbox]').checked)?'on':'off'
+    }
+    let tag = t.dataset.fitPermGroupType;
+    let slug = t.dataset.courseSlug;
+    if (value) {
+        post(`/course/${slug}/edit/options/${tag}/${value}`);
+    }
+  });
+
+  delegate('#change-slug', 'click', (e, t) => {
+    let slug = t.dataset.courseSlug;
+    let value = document.querySelector('#course-slug').value;
+    post(`/course/${slug}/edit/slug`, {course_slug: value}, (responseText, xhr) => {
+        if (xhr.status == 200) {
+            window.location.href = `/course/${JSON.parse(xhr.response)['slug']}/edit`
+        } else {
+                console.log('DEV: give me some message')
+            }
+    });
+  });
+
+  delegate('a[data-course-edit-remove-teacher]', 'click', (e, t) => {
+    let teacherId = e.target.dataset.teacherId;
+    let slug = t.dataset.courseSlug;
+    post(`/course/${slug}/edit/remove/teacher/${teacherId}`, {}, (responseText, xhr) => {
+        let responseJSON = JSON.parse(xhr.response);
+        let alert = $('#add-teacher-alert');
+        if (xhr.status == 200) {
+            let teacherDiv = $(`#teacher-${responseJSON['teacher_id']}`);
+            if (teacherDiv) {
+                teacherDiv.remove()
+            }
+        } else {
+                alert.css("display", "block");
+                alert.removeClass('alert-success');
+                alert.addClass('alert-danger');
+                alert.html(responseJSON.message);
+            }
+    });
+  });
+
+  delegate('#add-teacher', 'click', (e, t) => {
+    e.preventDefault();
+    let email = document.querySelector('#add-teacher-email');
+    let teacherId = e.target.dataset.teacherId;
+    let slug = t.dataset.courseSlug;
+    post(`/course/${slug}/edit/add/teacher`, {teacher_email: email.value}, (responseText, xhr) => {
+        let alert = $('#add-teacher-alert');
+        let responseJSON = JSON.parse(xhr.response);
+        alert.css("display", "block");
+        if (xhr.status == 400) {
+            alert.removeClass('alert-success');
+            alert.addClass('alert-danger');
+        } else {
+            alert.addClass('alert-success');
+            alert.removeClass('alert-danger');
+            $('#teachers-list').append(
```

Comment:
We definitely don't want to dump a bunch of HTML in our JavaScript.

This is better with AJAX partial templates or hiding/showing existing divs.

We might just fix this later, though.